pkg/detectors/aws/aws.go

```diff
@@ -115,12 +131,19 @@ func (s scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				Raw:          []byte(resIDMatch),
 				Redacted:     resIDMatch,
 				RawV2:        []byte(resIDMatch + resSecretMatch),
+				ExtraData: map[string]string{
+					"resource_type": resourceTypes[resIDMatch[0:4]],
```

Comment:
question: Instead of this slice indexing did you consider adding a capture group to the prefix and just using that directly. That should prevent any potential IOOR panics.
For example:
New regex - `idPat     = regexp.MustCompile(`\b(AKIA|ABIA|ACCA|ASIA)([0-9A-Z]{16})\b`)`
Then within the code around matching:
```
for _, idMatch := range idMatches {
		if len(idMatch) != 3 {
			continue
		}
		resIDMatch := strings.TrimSpace(idMatch[0])

		....

		for _, secretMatch := range secretMatches {
			if len(secretMatch) != 2 {
				continue
			}
			resSecretMatch := strings.TrimSpace(secretMatch[1])

			s1 := detectors.Result{
				DetectorType: detectorspb.DetectorType_AWS,
				Raw:          []byte(resIDMatch),
				Redacted:     resIDMatch,
				RawV2:        []byte(resIDMatch + resSecretMatch),
				ExtraData: map[string]string{
					"resource_type": resourceTypes[idMatch[1]],  // HERE WE JUST USE THE CAPTURE
				},
			}
...
```

Reply:
Just updated with latest commit.